pyo3-macros-backend/src/utils.rs

```diff
@@ -67,14 +68,58 @@ pub fn option_type_argument(ty: &syn::Type) -> Option<&syn::Type> {
     None
 }
 
+// TODO: Replace usage of this by [`syn::LitCStr`] when on MSRV 1.77
+#[derive(Clone)]
+pub struct LitCStr {
+    lit: CString,
+    span: Span,
+    #[allow(dead_code)]
```

Comment:
Maybe cfg this so we remember to clean up later?

```suggestion
    #[cfg_attr(c_str_lit, allow(dead_code))]
```